code/Control/UserDefinedFormController.php

```diff
@@ -135,6 +135,10 @@ protected function addUserFormsValidatei18n()
      */
     public function index(HTTPRequest $request = null)
     {
+        // If $this is not a subclass of UserDefinedForm
+        if (!$this->dataRecord instanceof UserDefinedForm) {
```

Comment:
This breaks userforms for any other association. This page type is not the only manner in which they can be applied.
[Elemental](https://github.com/dnadesign/silverstripe-elemental-userforms), for example.

Reply:
I personally did not have any trouble when testing this with the elemental userforms.